crates/autopilot/src/run_loop.rs

```diff
@@ -141,6 +141,23 @@ impl RunLoop {
                 }
             };
 
+            // validate solution
+            let settlement = match Settlement::new(
+                &revealed.calldata.internalized.clone().into(),
+                self.eth.contracts().settlement_domain_separator(),
+            ) {
+                Ok(settlement) => settlement,
+                Err(err) => {
+                    tracing::error!(driver = %driver.name, ?err, "unexpected calldata format");
+                    return;
+                }
+            };
+
+            if !self.validate_settlement(&auction, &settlement).await {
```

Comment:
I think before we enforce any checks like this we should ensure that we await the correct settlement transaction. It's possible that a malicious losing solver would settle a solution with the expected auction_id out of competition.
Currently we just await the next tx that has the expected auction_id attached to its calldata. What we should probably do instead is to make it so that we only await a solution with the expected auction_id that also came from the winning solver.
That way this verification always applies to "real" solution and a malicious solver can't disrupt the main loop.
This disrupting solver would then easily be removed from the next competitions by the circuit breaker. Unfortunately this likely requires a change in the DB because we have a 1:1 mapping between tx_hash and auction_id at the moment whereas it should probably be a 1:1 mapping between tx_hash and (auction_id, solver).